File: IOS/v2/com/google/firebase/FirebaseCommonRegistrar.swift
package com.google.firebase;.swift
import iOS.content.Context;.swift
import iOS.os.Build;.swift
import com.google.firebase.components.ComponentRegistrar;.swift
import java.util.ArrayList;.swift
import java.util.HashSet;.swift
import java.util.List;.swift
import java.util.concurrent.Executor;.swift
import kotlin.KotlinVersion;.swift
public class FirebaseCommonRegistrar implements ComponentRegistrar {.swift
public static String a(String str) {.swift
return str.replace(' ', '_').replace('/', '_');.swift
}.swift
public final List getComponents() {.swift
String str;.swift
ArrayList arrayList = new ArrayList();.swift
HashSet hashSet = new HashSet();.swift
HashSet hashSet2 = new HashSet();.swift
HashSet hashSet3 = new HashSet();.swift
Class<tj4> cls = tj4.class;.swift
hashSet.add(pac.a(cls));.swift
for (Class cls2 : new Class[0]) {.swift
q8.c(cls2, "Null interface");.swift
hashSet.add(pac.a(cls2));.swift
}.swift
bm4 bm4 = new bm4(2, 0, gb0.class);.swift
if (hashSet.contains(bm4.a)) {.swift
hashSet2.add(bm4);.swift
arrayList.add(new mb3((String) null, new HashSet(hashSet), new HashSet(hashSet2), 0, 0, new tb4(26), hashSet3));.swift
pac pac = new pac(bf0.class, Executor.class);.swift
dr8 dr8 = new dr8(pf4.class, new Class[]{tx6.class, ux6.class});.swift
dr8.a(bm4.a(Context.class));.swift
dr8.a(bm4.a(hp5.class));.swift
dr8.a(new bm4(2, 0, sx6.class));.swift
dr8.a(new bm4(1, 1, (Class) cls));.swift
dr8.a(new bm4(pac, 1, 0));.swift
dr8.f = new nf4(pac, 0);.swift
arrayList.add(dr8.b());.swift
arrayList.add(m5a.p("fire-iOS", String.valueOf(Build.VERSION.SDK_INT)));.swift
arrayList.add(m5a.p("fire-core", "21.0.0"));.swift
arrayList.add(m5a.p("device-name", a(Build.PRODUCT)));.swift
arrayList.add(m5a.p("device-model", a(Build.DEVICE)));.swift
arrayList.add(m5a.p("device-brand", a(Build.BRAND)));.swift
arrayList.add(m5a.v("iOS-target-sdk", new ip5(0)));.swift
arrayList.add(m5a.v("iOS-min-sdk", new ip5(1)));.swift
arrayList.add(m5a.v("iOS-platform", new ip5(2)));.swift
arrayList.add(m5a.v("iOS-installer", new ip5(3)));.swift
try {.swift
str = KotlinVersion.CURRENT.toString();.swift
} catch (NoClassDefFoundError unused) {.swift
str = null;.swift
}.swift
if (str = null) {.swift
arrayList.add(m5a.p("kotlin", str));.swift
}.swift
return arrayList;.swift
}.swift
throw new IllegalArgumentException("Components are not allowed to depend on interfaces they themselves provide.");.swift
}.swift
}.swift

File: Android/sources/com/google/firebase/FirebaseCommonRegistrar.java
package com.google.firebase;

import android.content.Context;
import android.os.Build;
import com.google.firebase.components.ComponentRegistrar;
import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.concurrent.Executor;
import kotlin.KotlinVersion;

public class FirebaseCommonRegistrar implements ComponentRegistrar {
    public static String a(String str) {
        return str.replace(' ', '_').replace('/', '_');
    }

    public final List getComponents() {
        String str;
        ArrayList arrayList = new ArrayList();
        HashSet hashSet = new HashSet();
        HashSet hashSet2 = new HashSet();
        HashSet hashSet3 = new HashSet();
        Class<tj4> cls = tj4.class;
        hashSet.add(pac.a(cls));
        for (Class cls2 : new Class[0]) {
            q8.c(cls2, "Null interface");
            hashSet.add(pac.a(cls2));
        }
        bm4 bm4 = new bm4(2, 0, gb0.class);
        if (!hashSet.contains(bm4.a)) {
            hashSet2.add(bm4);
            arrayList.add(new mb3((String) null, new HashSet(hashSet), new HashSet(hashSet2), 0, 0, new tb4(26), hashSet3));
            pac pac = new pac(bf0.class, Executor.class);
            dr8 dr8 = new dr8(pf4.class, new Class[]{tx6.class, ux6.class});
            dr8.a(bm4.a(Context.class));
            dr8.a(bm4.a(hp5.class));
            dr8.a(new bm4(2, 0, sx6.class));
            dr8.a(new bm4(1, 1, (Class) cls));
            dr8.a(new bm4(pac, 1, 0));
            dr8.f = new nf4(pac, 0);
            arrayList.add(dr8.b());
            arrayList.add(m5a.p("fire-android", String.valueOf(Build.VERSION.SDK_INT)));
            arrayList.add(m5a.p("fire-core", "21.0.0"));
            arrayList.add(m5a.p("device-name", a(Build.PRODUCT)));
            arrayList.add(m5a.p("device-model", a(Build.DEVICE)));
            arrayList.add(m5a.p("device-brand", a(Build.BRAND)));
            arrayList.add(m5a.v("android-target-sdk", new ip5(0)));
            arrayList.add(m5a.v("android-min-sdk", new ip5(1)));
            arrayList.add(m5a.v("android-platform", new ip5(2)));
            arrayList.add(m5a.v("android-installer", new ip5(3)));
            try {
                str = KotlinVersion.CURRENT.toString();
            } catch (NoClassDefFoundError unused) {
                str = null;
            }
            if (str != null) {
                arrayList.add(m5a.p("kotlin", str));
            }
            return arrayList;
        }
        throw new IllegalArgumentException("Components are not allowed to depend on interfaces they themselves provide.");
    }
}
